api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
How about:

```
Context {
  Requested: {
    State: Continue
  },
  Status: {
    State: Paused,
    Thread: { ... },
    BreakPoint: { ... },
  }
}
```

Breakpoints and threads would be requested through their own endpoints:

```go
type BreakPoint struct {
	ID           int    `json:"id"`
	FunctionName string `json:"functionName"`
	File         string `json:"file"`
	Line         int    `json:"line"`
	Addr         uint64 `json:"addr"`
}
```
```go
type Thread struct {
	ID          int     `json:"id"`
	Status      uint32  `json:"status"`
	CurrentPC   uint64  `json:"currentPc"`
	CurrentLine *PCLine `json:"currentLine,omitempty"`
	IsCurrent   bool    `json:"isCurrent,omitempty"`
}
```